src/interpreter/ByteCode.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2022-present Samsung Electronics Co., Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Walrus.h"
+
+#include "interpreter/ByteCode.h"
+
+namespace Walrus {
+
+// Return static size of opcodes
+size_t ByteCode::getBaseSize(OpcodeKind opcode)
+{
+    ASSERT(g_byteCodeInfo[opcode].m_size > 0);
+
+    return g_byteCodeInfo[opcode].m_size;
+}
+
+// Returns real size of generated opcodes
+size_t ByteCode::getSize(FunctionType* currentFunction, Module* module, Vector<ModuleFunction*> funcs, Vector<TagType*> tagTypes)
```

Comment:
I ma thinking how could we make this nicer. What about creating two functions from it, one for the parser inside the parser. The other (public) function has the `module` argument.

Reply:
I have reworked it. The one that the parser uses is only defined if NDEBUG is defined, because it is only used in `dumpByteCode(WASMParsingResult result)`. I have also added a getter for `m_tagTypes` to acces it from the module.